File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<int> InOut;
    
    if(s.front() == ')' || s.back() == '(') answer = false;
    else{
        for(auto c:s){
            if(c == '(')            InOut.push(1);
            else if(!InOut.empty()) InOut.pop();
        }
        answer = InOut.empty();
    }
    return answer;
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    cnt = 0
    
    if(s[0] == ')' or s[-1] == '('):
        answer = False
    else:
        for c in s:
            if c == '(':
                cnt += 1
            elif cnt>0 :
                cnt -= 1
        answer = (cnt == 0)

    return answer